public/components/panel/studioPanel/sampleDownloader.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import axios from 'axios'
+import { GridSampleObject } from '../workstationPanel/gridComponents'
+import './studioPanel.css'
+
+let SampleDownloaderMounted = false
+
+class SampleDownloader extends React.Component {
+  /**
+   * @param {{
+   * sample: GridSampleObject,
+   * audioContext: AudioContext,
+   * onComplete: (audioBuffer: AudioBuffer) => void,
+   * onError: () => void
+   * }} props
+   */
+  constructor(props) {
+    super(props)
+    this.state = {
+      downloadError: false,
+      downloadProgress: 0,
+      contentLength: 0.000001,
+    }
+  }
+
+  componentDidMount() {
+    const { sample } = this.props
+    SampleDownloaderMounted = true
+    this.startDownload(sample)
+  }
+
+  componentWillUnmount() {
+    SampleDownloaderMounted = false
+  }
+
+  /**
+   * @param {GridSampleObject} sample
+   */
+  startDownload = sample => {
+    const setProgress = progressPercentage => {
+      if (SampleDownloaderMounted) {
+        this.setState({ downloadProgress: progressPercentage })
+      }
+    }
+    axios({
+      responseType: 'arraybuffer',
+      url: 'https://cors-anywhere.herokuapp.com/' + sample.sampleSource,
```

Comment:
is this just some hardcoded URL until you have samples to download from or those this have significance?

Reply:
Shouldnt we store the download URL in the constants.js file too

Reply:
This is temp fix to allow us to download the test audio files from websites with blocking cors policy. It shouldn't be needed in the long run